components/movementsensor/gpsrtk/pmtk.go

```diff
@@ -0,0 +1,201 @@
+
+// I2CConfig is used for converting NMEA MovementSensor with RTK capabilities config attributes.
+type I2CConfig struct {
+	I2CBus      string `json:"i2c_bus"`
+	I2CAddr     int    `json:"i2c_addr"`
+	I2CBaudRate int    `json:"i2c_baud_rate,omitempty"`
+
+	NtripURL             string `json:"ntrip_url"`
+	NtripConnectAttempts int    `json:"ntrip_connect_attempts,omitempty"`
+	NtripMountpoint      string `json:"ntrip_mountpoint,omitempty"`
+	NtripPass            string `json:"ntrip_password,omitempty"`
+	NtripUser            string `json:"ntrip_username,omitempty"`
+}
+
+// Validate ensures all parts of the config are valid.
+func (cfg *I2CConfig) Validate(path string) ([]string, error) {
+	err := cfg.validateI2C(path)
+	if err != nil {
+		return nil, err
+	}
+
+	err = cfg.validateNtrip(path)
+	if err != nil {
+		return nil, err
+	}
+
+	return []string{}, nil
+}
+
+// validateI2C ensures all parts of the config are valid.
+func (cfg *I2CConfig) validateI2C(path string) error {
+	if cfg.I2CBus == "" {
+		return resource.NewConfigValidationFieldRequiredError(path, "i2c_bus")
+	}
+	if cfg.I2CAddr == 0 {
+		return resource.NewConfigValidationFieldRequiredError(path, "i2c_addr")
+	}
+	return nil
+}
+
+// validateNtrip ensures all parts of the config are valid.
+func (cfg *I2CConfig) validateNtrip(path string) error {
+	if cfg.NtripURL == "" {
+		return resource.NewConfigValidationFieldRequiredError(path, "ntrip_url")
+	}
+	return nil
+}
+
+func init() {
+	resource.RegisterComponent(
+		movementsensor.API,
+		i2cRtkmodel,
+		resource.Registration[movementsensor.MovementSensor, *I2CConfig]{
+			Constructor: newRTKI2C,
+		})
+}
+
+func newRTKI2C(
+	ctx context.Context,
+	deps resource.Dependencies,
+	conf resource.Config,
+	logger logging.Logger,
+) (movementsensor.MovementSensor, error) {
+	return makeRTKI2C(ctx, deps, conf, logger, nil)
+}
+
+// makeRTKI2C is separate from newRTKI2C, above, so we can pass in a non-nil mock I2C bus during
+// unit tests.
+func makeRTKI2C(
+	ctx context.Context,
+	_ resource.Dependencies,
+	conf resource.Config,
+	logger logging.Logger,
+	mockI2c buses.I2C,
+) (movementsensor.MovementSensor, error) {
+	newConf, err := resource.NativeConfig[*I2CConfig](conf)
+	if err != nil {
+		return nil, err
+	}
+
+	cancelCtx, cancelFunc := context.WithCancel(context.Background())
+	g := &gpsrtk{
+		Named:      conf.ResourceName().AsNamed(),
+		cancelCtx:  cancelCtx,
+		cancelFunc: cancelFunc,
+		logger:     logger,
+		err:        movementsensor.NewLastError(1, 1),
+	}
+
+	ntripConfig := &gpsutils.NtripConfig{
+		NtripURL:             newConf.NtripURL,
+		NtripUser:            newConf.NtripUser,
+		NtripPass:            newConf.NtripPass,
+		NtripMountpoint:      newConf.NtripMountpoint,
+		NtripConnectAttempts: newConf.NtripConnectAttempts,
+	}
+
+	g.ntripClient, err = gpsutils.NewNtripInfo(ntripConfig, g.logger)
+	if err != nil {
+		return nil, err
+	}
+	g.InputProtocol = "i2c"
+
+	config := gpsutils.I2CConfig{
```

Comment:
super nit, I'd rather this be called i2cConf so it doesn't potentially ever shadow the `go.viam.com/rdk/config` package. i don't think we'll ever use it in this driver but just in case.